### ISSUE TITLE
pass generate api stream is undefined

### ISSUE THREAD

**Issue by nikhilenmudi** [alexandercerutti/passkit-generator#18]
Hi,
I am getting undefined stream on using the following code. I dont see any error in my catch either. Please help as I am relatively new to this.
The stream is undefined. 

```javascript
app.get('/downloadpass', async (req, res, next) => {
 let examplePass = new Pass({
     model: "./passModels/Coupon",
     certificates: {
         wwdr: "./certs/wwdr.pem",
         signerCert: "./certs/signerCert.pem",
         signerKey: {
             keyFile: "./certs/passkey.pem",
             passphrase: "12345"
         }
     },
     overrides: {
         // keys to be added or overridden
         serialNumber: "AAGH44625236dddaffbda"
     },
     // if true, existing keys added through methods get overwritten
     // pushed in queue otherwise.
     shouldOverwrite: true
 });
  
 // Adding some settings to be written inside pass.json
 examplePass.barcode("36478105430"); // Random value
  
 // Generate the stream, which gets returned through a Promise
 try {
  let stream = await examplePass.generate();
  stream.pipe(res)
  
} catch (e) {
  //this will eventually be handled by your error handling middleware
  console.log(e) 
}
})
```

**Comment by alexandercerutti** [comment]
Hi @nikhilenmudi, have you tried to start your process with `DEBUG=*` before? Like `DEBUG=* node index.js`. Let me know.

**Comment by nikhilenmudi** [comment]
Never used the DEBUG.. i am running the app from command line as `node app.js` DEBUG is unrecognized

**Comment by alexandercerutti** [comment]
Yeah, so run it through the CLI with that command, as I said. `DEBUG=* node app.js`. 😃

**Comment by nikhilenmudi** [comment]
![image](https://user-images.githubusercontent.com/6737821/56928877-1de25b80-6aa6-11e9-9077-2d88895a087d.png)

**Comment by alexandercerutti** [comment]
Oh okay, you are on Windows. That's why.
Try with `set DEBUG=* & node app.js`. (source: [Debug Readme](https://github.com/visionmedia/debug#windows-command-prompt-notes))
Once done, you'll be able to unset it as `set DEBUG=` (with nothing after the equal symbol).

**Comment by nikhilenmudi** [comment]
![image](https://user-images.githubusercontent.com/6737821/56929180-f50e9600-6aa6-11e9-9ebd-1f81cef7b01c.png)

**Comment by nikhilenmudi** [comment]
The error i am getting is cannot read pipe of undefined. Not sure why the stream is undefined. I have tried printing it too, also tried with generate().then(stream => { stream.pipe(res}) but same issue.

**Comment by alexandercerutti** [comment]
Because also other packets, like `expressjs` use DEBUG to print stuff.
Anyway, no important information is provided... that's weird.
Sadly I don't have enough infos to understand what's going on; I can go only by attempts.
Have you tried to follow the examples I created and check if the info you inserted in configuration works also with them?

**Comment by nikhilenmudi** [comment]
No worries. Thanks for helping out. Which examples are you asking about? 
Does my express endpoint code look fine? I havent used await/async stuff much, is the code correct? May be the stream is not ready and I am piping it out before?

**Comment by alexandercerutti** [comment]
The express should not be a problem at all. I'm talking about the examples available [here](https://github.com/alexandercerutti/passkit-generator/tree/master/examples). Please read the README to be sure to run them correctly. Copy your configuration files (like keys, model and so on) to the cloned project and run whatever example you want but `fields.js`, because it has an hardcoded model. Therefore, whatever model you specify at the endpoint in README.md, will be redirected to `exampleBooking.pass`.

The code runs, async/await are correct.
The stream should work correctly. If you look in the examples I talked about before, you'll see I'm doing exactly the same thing you are doing. 👍

Keep me update! Thank you!

**Comment by ndrozd** [comment]
Same for me 
`TypeError: Cannot read property 'pipe' of undefined
    at /Users/nikitadrozd/Documents/projects/passlit/passkit-generator/examples/fields.js:154:10
`

**Comment by alexandercerutti** [comment]
One can be a user-related problem. Two make it start to begin a real problem.
Can I ask you to test with the previous version? @ndrozd @nikhilenmudi

```sh
$ npm install passkit-generator@1.6.0
```

This problem might be related to the fact I changed the flow from Promise to async/await and I might not have encountered the problems you are experiencing.

**Comment by ndrozd** [comment]
Same with 1.5.9 :(

**Comment by ndrozd** [comment]
1.5.9 works okay. 
Should examples there return error like this?
`Error: Invalid certificate loaded. WWDR.pem does not exist.`

**Comment by alexandercerutti** [comment]
What about v.1.6.0 instead?
The error is returned by the package if one of the certificates path does not exist.

**Comment by ksaine** [comment]
You should add "const forge = require("node-forge");" in src/utils.js. This is desolve the problev.

**Comment by alexandercerutti** [comment]
@ksaine have you tried? I probably forgot when I separated the functions in `utils.js`.
I'll commit, push and publish it this evening if this is the problem. @ndrozd @nikhilenmudi have you tried this solution?

**Comment by ksaine** [comment]
Yes, of course. I had the same problem.

**Comment by alexandercerutti** [comment]
I was also noticing that I forgot add a throw err in the catch if the error is not an `ENOENT`. So probably for that reason it was not showing any error. It was catching everything correctly.

**Comment by nikhilenmudi** [comment]
I found the problem when i used the @destinationstransfers/passkit version. It  gave some errors related to having missing cert and not matching passtypeid. Yes you are right, I think in your implementation the generate needs to throw error

**Comment by alexandercerutti** [comment]
@nikhilenmudi if you see [this commit](https://github.com/alexandercerutti/passkit-generator/commit/18893ac7421518d2cf1d404060ccda7ceea145c5?diff=split#diff-6e62d615f6212dc8cd5e6a0bf0f4eb93L72), you can see that there was a throw that I forgot to re-introduce when I moved to async/await.

Oh, awesome 😂. So both version was not working for different reasons. 😂

**Comment by alexandercerutti** [comment]
Okay, @nikhilenmudi @ndrozd and @ksaine! I've published version 1.6.2 with the fixes we talked about. I tried again before putting my hands on and it gave me the same error that I did not saw the first time.
Be sure to update it.

Thank you very much for using and helped me to improve this package. ❤
If you find any other issues, feel free to open other issue topics and I'll be happy to help you!
